src/components/info/warning.js

```diff
@@ -0,0 +1,57 @@
+import React, { Suspense, lazy } from "react";
+import { connect } from "react-redux";
+import styled from 'styled-components';
+import { FaExclamationTriangle } from "react-icons/fa";
+import { withTranslation } from 'react-i18next';
+
+const MarkdownDisplay = lazy(() => import("../markdownDisplay"));
+
+const WarningContainer = styled.div`
+  background-color: #FFF1D1;
+  padding: 10px;
+  display: flex;
+  align-items: center;
+  gap: 6px;
+  border-left: 6px solid rgb(255, 198, 65);
+`;
+
+const WarningIcon = styled.div`
+  font-size: 24px;
+  margin-top: 3px;
+`;
+
+const WarningText = styled.div`
+  font-size: 15px;
+`;
+
+/**
+ * React component for the warning of the current dataset.
+ */
+@connect((state) => {
+  return {
+    metadata: state.metadata
```

Comment:
In general it's best to access state in the most fine-grained way possible to reduce unnecessary rendering. In this case it'd be `warning: state.metadata.warning`. Reality is a little more complex and depends on how the parent is rendered, but it's a good rule of thumb. 

(It's been a while since I was day-to-day wiring up React, but if you'd like me to track down specific docs let me know)

Reply:
Good tip, much cleaner. Updated:

https://github.com/nextstrain/auspice/blob/866829a0777f53dc799feee4a243ca431d98824f/src/components/info/warning.js#L36-L40